### PR TITLE
ENH: Add `affine_tolerance` flag to `MergeSeries`

### PR DESCRIPTION
**Pull request by mgxd**
This should help clear up the error found in https://github.com/nipreps/fmriprep/issues/2706

### CODE REVIEW

File: niworkflows/interfaces/nibabel.py [hunk 1]
```diff
@@ -186,6 +186,7 @@ class _MergeSeriesInputSpec(BaseInterfaceInputSpec):
     allow_4D = traits.Bool(
         True, usedefault=True, desc="whether 4D images are allowed to be concatenated"
     )
+    affine_tolerance = traits.Float(desc="Absolute tolerance allowed between image affines")
```
Comment (effigies) [inline niworkflows/interfaces/nibabel.py:189]: What about setting a default (taken from numpy), and doing our own check?

```suggestion
    affine_tolerance = traits.Float(1e-08, usedefault=True, desc="Absolute tolerance allowed between image affines")
```

File: niworkflows/interfaces/nibabel.py [hunk 2]
```diff
@@ -200,8 +201,17 @@ class MergeSeries(SimpleInterface):
...
         for f in self.inputs.in_files:
             filenii = nb.squeeze_image(nb.load(f))
+            if self.inputs.affine_tolerance:
+                if aff0 is None:
+                    aff0 = filenii.affine
+                elif not np.allclose(aff0, filenii.affine, atol=self.inputs.affine_tolerance):
+                    raise ValueError(
+                        "Difference in affines greater than allowed tolerance "
+                        f"{self.inputs.affine_tolerance}"
+                    )
```
Comment (effigies) [inline niworkflows/interfaces/nibabel.py:214]:
```suggestion
            if aff0 is None:
                aff0 = filenii.affine
            elif not np.allclose(aff0, filenii.affine, atol=self.inputs.affine_tolerance):
                raise ValueError(
                    "Difference in affines greater than allowed tolerance "
                    f"{self.inputs.affine_tolerance}"
                )
```

File: niworkflows/interfaces/nibabel.py [hunk 3]
```diff
@@ -214,7 +224,10 @@ def _run_interface(self, runtime):
                     "Input image has an incorrect number of dimensions" f" ({ndim})."
                 )
 
-        img_4d = nb.concat_images(nii_list)
+        img_4d = nb.concat_images(
+            nii_list,
+            check_affines=not bool(self.inputs.affine_tolerance)
+        )
```
Comment (effigies) [inline niworkflows/interfaces/nibabel.py:230]:
```suggestion
        img_4d = nb.concat_images(nii_list, check_affines=False)
```

**Review by effigies — APPROVED** [review]
Looks fine. I'd make the default threshold explicit.

**Comment by effigies** [comment]
@mgxd sorry to do this, but we're going to need this in `maint/1.4.x` to get it into fmriprep 21.0.x. Or do we just want to tell these users to upgrade to 22.0?

**Comment by mgxd** [comment]
Sounds good - I changed the PR to target 1.4.x

**Comment by effigies** [comment]
Thanks. It did pass, so I'm going to go ahead and merge.

**Pull request merged**